### PR TITLE
feat(`builder`): retry on incorrect host block error

### PR DESCRIPTION
**Pull request by Evalir**

### CODE REVIEW

File: crates/builder/src/tasks/submit.rs [hunk 1]
```diff
@@ -12,18 +12,28 @@ use alloy::signers::Signer;
...
+    SignRequest, SignResponse,
+    Zenith::{self, IncorrectHostBlock},
+};
 
 /// OAuth Audience Claim Name, required param by IdP for client credential grant
 const OAUTH_AUDIENCE_CLAIM: &str = "audience";
 
+pub enum ControlFlow {
+    Retry,
+    Skip,
```
Comment (prestwich) [inline crates/builder/src/tasks/submit.rs:33]: skip is never constructed. we need to convert things that would previously `?` or `return Err` to return `Ok(Skip)`

File: crates/builder/src/tasks/submit.rs [hunk 2]
```diff
@@ -156,6 +166,10 @@ impl SubmitTask {
                 "error in transaction submission"
             );
 
+            if e.message.contains(&hex::encode(IncorrectHostBlock::SELECTOR)) {
```
Comment (prestwich) [inline crates/builder/src/tasks/submit.rs:169]: use this:

https://github.com/alloy-rs/alloy/blob/3fc5943bd84c3d9883dc06118a0367a084668bdf/crates/json-rpc/src/response/error.rs#L343

**Review by rswanson — APPROVED** [review]

File: crates/builder/src/tasks/submit.rs [hunk 3]
```diff
@@ -218,8 +250,33 @@ impl SubmitTask {
...
+                                }
+                                tracing::error!("error handling inbound block: retrying");
+                                tokio::time::sleep(tokio::time::Duration::from_secs(2)).await;
+                            }
+                            Ok(ControlFlow::Skip) => {
+                                tracing::info!("skipping block");
+                                break;
+                            }
+                            Ok(ControlFlow::Done) => {
+                                tracing::info!("block landed successfully");
```
Comment (prestwich) [inline crates/builder/src/tasks/submit.rs:272]: need to set retries to 0 in Skip and Done arms

Comment (Evalir) [reply crates/builder/src/tasks/submit.rs:272]: pretty sure it gets reset on each iteration of the outer loop?

Comment (prestwich) [reply crates/builder/src/tasks/submit.rs:272]: oh you're right. i don't like the nested loop tho. kinda ugly and implies we should break out a function

Comment (Evalir) [reply crates/builder/src/tasks/submit.rs:272]: yep agreed, we can break it out in another PR

**Pull request merged**